code-studio/src/js/clientState.js

```diff
@@ -132,6 +132,24 @@ clientState.trackProgress = function (result, lines, testResult, scriptName, lev
 };
 
 /**
+ * Write down user progress for an entire script.
+ * @param {string} scriptName
+ * @param {Object<String, number>} progress
+ */
+clientState.batchTrackProgress = function (scriptName, progress) {
+  var data = {};
+  for (let level of Object.keys(progress)) {
+    if (progress[level] && progress[level] <= clientState.MAXIMUM_CACHABLE_RESULT) {
+      data[level] = progress[level];
```

Comment:
Progress of `0` and `null` are equivalent — [both map to 'not_tried'](https://github.com/code-dot-org/code-dot-org/blob/49e92752437c840bcca9a114979fcdd4f8d099b5/code-studio/src/js/progress.js#L23-L24).  I'll add a test [here](https://github.com/code-dot-org/code-dot-org/blob/a982cf356d61bf401b08d7965adbb50a08ad77be/code-studio/test/js/progressTest.js#L10) for the `null` case.